src/factors/Inertial/IMUDeltaFactor.jl

```diff
@@ -0,0 +1,467 @@
+using Manifolds
+using StaticArrays
+using Rotations
+using LinearAlgebra
+using DistributedFactorGraphs
+using Dates
+
+function TransformUtils.skew(v::SVector{3,T}) where T<:Real
+    return SMatrix{3,3,T}(
+            0,
+         v[3],
+        -v[2],
+        -v[3],  
+            0,  
+         v[1],
+         v[2],
+        -v[1],
+            0
+    )
+end
+
+struct IMUDeltaManifold <: AbstractManifold{ℝ} end
+
+# TODO maybe define the manifold like this
+# M = ProductManifold(TranslationGroup(3), TranslationGroup(3), SpecialOrthogonal(3), Euclidean(1))
+# M = ProductManifold(SpecialOrthogonal(3), TranslationGroup(3), TranslationGroup(3), Euclidean(1))
+# G = GroupManifold(M, MultiplicationOperation())
+
+#TODO 
+# also see SE₂(3) 
+
+const IMUDeltaGroup = GroupManifold{ℝ, IMUDeltaManifold, MultiplicationOperation}
+
+IMUDeltaGroup() = GroupManifold(IMUDeltaManifold(), MultiplicationOperation())
+
+Manifolds.manifold_dimension(::IMUDeltaManifold) = 9
+
+# Affine representation 
+# Δ = [ΔR Δv Δp;
+#      0   1 Δt;
+#      0   0  1] ⊂ \R^5x5
+
+#  ArrayPartition representation (TODO maybe swop order to [Δp; Δv; ΔR; Δt])
+# Δ = [ΔR; Δv; Δp; Δt] 
+
+# TODO is SemidirectProductGroup possible, Translation(3), Translation(3), SO(3), WhatAction?, Δt?
+
+function Manifolds.identity_element(M::IMUDeltaGroup) # was #SMatrix{5,5,Float64}(I)
+    ArrayPartition(
+        SMatrix{3,3,Float64}(I), # ΔR
+        @SVector(zeros(3)),      # Δv
+        @SVector(zeros(3)),      # Δp
+        0.0                      # Δt
+    )
+end
+
+DFG.getPointIdentity(M::IMUDeltaGroup) = identity_element(M)
+
+function Manifolds.affine_matrix(G::IMUDeltaGroup, p::ArrayPartition{T}) where T<:Real
+    return SMatrix{5,5,T}([
+        p.x[1]  p.x[2]  p.x[3];
+        0 0 0        1  p.x[4];
+        0 0 0        0       1
+    ])
```

Comment:
```suggestion
    return vcat(hcat(p.x[1], p.x[2], p.x[3]), @SMatrix [0 0 0 1 p.x[4]; 0 0 0 0 1])
```
This should be much faster.